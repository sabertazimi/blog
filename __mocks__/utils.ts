let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

const mockNavigatorClipboard = () => {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(global.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
