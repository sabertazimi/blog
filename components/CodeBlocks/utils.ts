import parseNumericRange from 'parse-numeric-range'
import type { Language } from 'prism-react-renderer'

const normalizeCode = (code: string = '') => code.replace(/\n+$/, '')

function isLanguage(language: string): language is Language {
  return language.length > 0
}

function normalizeLanguage(languageClass?: string): Language {
  if (languageClass != null) {
    const language = languageClass.replace('language-', '')

    if (isLanguage(language))
      return language
  }

  return 'typescript'
}

function normalizeLanguageName(language: string): string {
  switch (language) {
    case 'html':
      return 'HTML'
    case 'xml':
      return 'XML'
    case 'yml':
    case 'yaml':
      return 'YAML'
    case 'css':
      return 'CSS'
    case 'json':
      return 'JSON'
    case 'md':
    case 'markdown':
      return 'Markdown'
    case 'js':
    case 'javascript':
      return 'JavaScript'
    case 'ts':
    case 'typescript':
      return 'TypeScript'
    case 'coffee':
    case 'coffeescript':
      return 'CoffeeScript'
    case 'jsx':
    case 'tsx':
      return 'React'
    case 'objc':
    case 'objectivec':
      return 'Objective-C'
    default:
      return language.charAt(0).toUpperCase() + language.slice(1)
  }
}

function normalizeLines(expression: string): Set<number> {
  return new Set(parseNumericRange(expression))
}

export {
  normalizeCode,
  normalizeLanguage,
  normalizeLanguageName,
  normalizeLines,
}
