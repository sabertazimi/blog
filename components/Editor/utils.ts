function languageToFilepath(language?: string): string {
  switch (language) {
    case 'css':
      return '/styles.css'
    case 'js':
    case 'javascript':
      return '/App.js'
    case 'ts':
    case 'typescript':
      return '/App.ts'
    case 'jsx':
      return '/App.jsx'
    case 'tsx':
      return '/App.tsx'
    case 'vue':
      return '/src/App.vue'
    case undefined:
    default:
      return '/App.tsx'
  }
}

function normalizeFilepath(filename?: string, language?: string): string {
  if (filename == null || filename === '')
    return languageToFilepath(language)

  if (!filename.startsWith('/'))
    return `/${filename}`

  return filename
}

export { normalizeFilepath }
