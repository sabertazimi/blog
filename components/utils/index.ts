import classNames from 'classnames';

export { classNames };
