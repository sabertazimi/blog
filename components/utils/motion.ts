export { AnimatePresence, motion } from 'framer-motion';
export type { MotionProps } from 'framer-motion';
