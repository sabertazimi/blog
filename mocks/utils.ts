let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

function mockNavigatorClipboard() {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(globalThis.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
