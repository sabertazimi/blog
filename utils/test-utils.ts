import { render as testingRender } from '@testing-library/react'
import { ConfigProvider } from 'antd'
import { createElement } from 'react'

export * from '@testing-library/react'

export function render(ui: Parameters<typeof testingRender>[0], options?: Parameters<typeof testingRender>[1]): ReturnType<typeof testingRender> {
  return testingRender(
    createElement(
      ConfigProvider,
      {
        theme: {
          hashed: false,
        },
      },
      ui,
    ),
    options,
  )
}
