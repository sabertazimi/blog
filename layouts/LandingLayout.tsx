import { MetaHeader } from '@components'
import { cx } from '@components/utils'
import landingImage from '@images/landing.jpg'
import type { ReactNode } from 'react'

interface Props {
  children: ReactNode
}

const LandingLayout = ({ children }: Props): JSX.Element => (
  <div>
    <MetaHeader />
    <div
      className={cx(
        'flex-container m-0 h-screen w-full flex-col overflow-y-auto overflow-x-hidden',
        'bg-cover bg-center bg-no-repeat p-0 text-light'
      )}
      style={{
        backgroundImage: `url("${landingImage.src}")`,
      }}
    >
      {children}
    </div>
  </div>
)

export default LandingLayout
