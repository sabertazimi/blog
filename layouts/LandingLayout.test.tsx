import { render } from '@/utils'
import LandingLayout from './LandingLayout'

describe('LandingLayout', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(
      <LandingLayout>
        <div>LandingLayout</div>
      </LandingLayout>,
    )

    expect(container).toMatchSnapshot()
  })
})
