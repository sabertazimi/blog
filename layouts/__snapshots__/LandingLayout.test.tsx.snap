// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingLayout should render correctly (snapshot) 1`] = `
<div>
  <div>
    <div
      class="flex-container m-0 h-screen w-full flex-col overflow-y-auto overflow-x-hidden bg-cover bg-center bg-no-repeat p-0 text-light"
      style="background-image: url(/img.jpg);"
    >
      <div>
        LandingLayout
      </div>
    </div>
  </div>
</div>
`;
